overlord/devicestate/devicemgr.go

```diff
@@ -329,22 +333,81 @@ func (m *DeviceManager) maybeSetupUbuntuSave() error {
 	if !runMntSaveMounted {
 		// we don't have ubuntu-save, save will be used directly
 		logger.Noticef("no ubuntu-save mount")
-		m.saveAvailable = true
 		return nil
 	}
 
-	logger.Noticef("bind-mounting ubuntu-save under %v", dirs.SnapSaveDir)
+	sysd := systemd.New(systemd.SystemMode, progress.Null)
+
+	// In newer core20/core22 we have a mount unit for ubuntu-save, which we
+	// will try to start first. Invoking systemd-mount in this case would fail.
+	err = sysd.Enable([]string{"var-lib-snapd-save.mount"}, systemd.EnableOptions{
+		StartNow: true,
+	})
```

Comment:
can you explain why this is an enable and not a just start, I'm missing something, as the comment mentions starting the unit

Reply:
It actually didn't occur to me that I was using Enable instead of Start. I had convinced myself it was neccessary to enable and start it instead of just trying to start it. I tested using only Start instead of Enable and that works just fine. Thanks for catching this. I changed this to a start command and updated the unit tests.

I already did some changes to Enable/EnableNoReload, should I undo those are those fine to keep?

Reply:
The key question is whether you just want to start the unit now (then it should only be a Start), or if you'd like it to be started automatically at the next boots too (for this you need Enable).

Reply:
> The key question is whether you just want to start the unit now (then it should only be a Start), or if you'd like it to be started automatically at the next boots too (for this you need Enable).

When the system is in run mode the mount unit is presented as enabled - which means it seems to be enabled already. I think the only issue is that it fails to start due to the missing path which is required in its mount unit.
`ConditionPathIsMountPoint=/run/mnt/ubuntu-save`

So during install mode it's not started, but the moment we reboot into run mode it's automatically started up. We have tests in core-base/core20 that verify this